contracts/peripheral/YieldDirectorV2.sol

```diff
@@ -0,0 +1,488 @@
+    */
+    function addToSohmDeposit(uint256 depositId_, uint256 amount_) external {
+        uint256 gohmAmount = _toAgnostic(amount_);
+        _increaseDeposit(depositId_, gohmAmount);
+
+        IERC20(sOHM).safeTransferFrom(msg.sender, address(this), amount_);
+        IERC20(sOHM).approve(address(staking), amount_);
+        staking.wrap(address(this), amount_);
+    }
+
+    /**
+        @notice Withdraw donor's gOHM from vault
+        @param depositId_ Deposit ID to remove gOHM deposit from
+        @param amount_ Amount of gOHM deposit to remove and return to donor
+    */
+    function withdrawPrincipal(uint256 depositId_, uint256 amount_) external {
+        uint256 amountWithdrawn = _withdraw(depositId_, amount_);
+
+        IERC20(gOHM).safeTransfer(msg.sender, amountWithdrawn);
+    }
+
+    /**
+        @notice Withdraw donor's gOHM from vault, and return it as sOHM
+        @param depositId_ Deposit ID to remove gOHM debt from
+        @param amount_ Amount of gOHM debt to remove and return to donor as sOHM
+    */
+    function withdrawPrincipalAsSohm(uint256 depositId_, uint256 amount_) external {
+        uint256 amountWithdrawn = _withdraw(depositId_, amount_);
+
+        IERC20(sOHM).approve(address(staking), amountWithdrawn);
+        staking.unwrap(msg.sender, amountWithdrawn);
+    }
+
+    /**
+        @notice Withdraw all gOHM from all donor positions
+    */
+    function withdrawAll() external {
+        if (withdrawDisabled) revert YieldDirector_WithdrawalsDisabled();
+
+        uint256[] memory depositIds = depositorIds[msg.sender];
+
+        uint256 depositsLength = depositIds.length;
+        if (depositsLength == 0) revert YieldDirector_NoDeposits();
+
+        uint256 principalTotal = 0;
+
+        for (uint256 index = 0; index < depositsLength; ++index) {
+            DepositInfo storage currDeposit = depositInfo[depositIds[index]];
+
+            principalTotal += currDeposit.principalAmount;
+
+            _withdrawAllPrincipal(depositIds[index]);
+        }
+
+        IERC20(gOHM).safeTransfer(msg.sender, _toAgnostic(principalTotal));
+
+        emit AllWithdrawn(msg.sender, _toAgnostic(principalTotal));
+    }
+
+    /**
+        @notice Get deposited gOHM amounts for specific recipient (updated to current index
+                based on sOHM equivalent amount deposit)
+        @param donor_ Address of user donating yield
+        @param recipient_ Address of user receiving donated yield
+    */
+    function depositsTo(address donor_, address recipient_) external view returns (uint256) {
+        uint256[] storage depositIds = depositorIds[donor_];
+
+        for (uint256 index = 0; index < depositIds.length; ++index) {
+            uint256 id = depositIds[index];
+
+            if (recipientLookup[id] == recipient_) {
+                return _toAgnostic(depositInfo[id].principalAmount);
+            }
+        }
+
+        return 0;
```

Comment:
Should this emit an error instead?   Would indicate that the recipient was not found